src/cli/command_processor.py

```diff
@@ -14,71 +14,117 @@ def __init__(self, remote_api: RemoteApi, local_api: LocalApi):
-        pass
+        self.__remote.login(username=username, password=password, host=host)
 
     @log_function
     def logout(self, host: str) -> None:
-        pass
+        self.__remote.logout(host=host)
 
     @log_function
     def applications_create(
         self, application: str, roles: List[str], path: Path
     ) -> None:
-        pass
+        self.__local.create_application(application, roles, path)
 
     @log_function
     def applications_init(self, path: Path) -> None:
-        pass
+        self.__local.init_application(path)
 
     @log_function
     def applications_upload(self, application: str) -> None:
-        pass
+        self.__remote.upload_application(application)
 
     @log_function
     def applications_list(self, remote: bool, local: bool) -> List[ApplicationType]:
-        return []
+        app_list = []
+        if remote:
+            app_list.extend(self.__remote.list_applications())
+
+        if local:
+            app_list.extend(self.__local.list_applications())
+
+        return app_list
 
     @log_function
     def applications_delete(self, application: str) -> None:
-        pass
+        self.__remote.delete_application(application)
 
     @log_function
     def applications_publish(self, application: str) -> None:
-        pass
+        self.__remote.publish_application(application)
 
     @log_function
-    def applications_validate(self, application: str) -> None:
-        pass
+    def applications_validate(self, application: str) -> Tuple[bool, str]:
+        return self.__local.is_application_valid(application)
 
     @log_function
-    def experiments_create(self, name: str, application: str, local: bool) -> None:
-        pass
+    def experiments_create(self, name: str, application: str, network: str, local: bool, path: Path) \
+        -> Tuple[bool, str]:
+        if local:
+            app_config = self.__local.get_application_config(application)
+
+            if self.__local.check_valid_network(network, app_config):
+                return self.__local.create_experiment(name=name, app_config=app_config, network=network, path=path,
+                                                      application=application)
+
+            return False, f"The specified network '{network}' does not exist."
+
+        return False, 'Remote experiment creation is not yet enabled.'
 
     @log_function
     def experiments_delete(self, path: Path) -> None:
-        pass
+        self.__remote.delete_experiment(path)
+        self.__local.delete_experiment(path)
 
     @log_function
     def __is_application_local(self, application: str) -> bool:
         pass
 
     @log_function
     def experiments_run(self, path: Path, block: bool) -> Optional[List[ResultType]]:
-        return []
+
+        results = None
+        is_local = self.__local.is_experiment_local(path)
+
+        if is_local:
+            results = self.__local.run_experiment(path, block)
```

Comment:
Running locally will always be blocked. We only have one simulator for this job. How do you interpret a non-blocked local run?